lib/persist-state.c

```diff
@@ -174,6 +176,42 @@ _wait_until_map_release(PersistState* self)
 }
 
 static gboolean
+pwrite_strict(gint fd, const void *buf, size_t count, off_t offset)
+{
+  ssize_t written = pwrite(fd, buf, count, offset);
+  gboolean result = TRUE;
```

Comment:
since we have to touch the patch anyway (because of the circular dependency), I would mention a nitpick here:

we shouldn't use pwrite_strict() as a function name. What is strict about it? I am not even sure that if we simply inlined the pwrite() call to increase_file_size() we would be worse off. we are wrapping a single function call, with a _very_ specific error translation which only makes sense from the single call site we are using it.

it would look like this if we did it in increase_file_size():

```
gint rc;

rc = pwrite(self->fd, pad_buffer, length, self->current_size);
if (rc < 0 || rc != length)
  {
     msg_error(...);
  }

```

The errno mapping part can become like this:

evt_tag_string("error", rc < 0 ? g_strerror(errno) : "short write");

what do you think?